go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -420,8 +420,12 @@ func (oc *SecondaryLayer3NetworkController) newRetryFramework(
 // Start starts the secondary layer3 controller, handles all events and creates all needed logical entities
 func (oc *SecondaryLayer3NetworkController) Start(ctx context.Context) error {
 	klog.Infof("Start secondary %s network controller of network %s", oc.TopologyType(), oc.GetNetworkName())
-
-	if err := oc.Init(ctx); err != nil {
+	networkID, err := oc.getNetworkID()
+	if err != nil {
+		return fmt.Errorf("unable to set networkID on secondary L3 controller for network %s", oc.GetNetworkName())
+	}
+	oc.networkID = &networkID
```

Comment:
why do we need to set networkID here ? Looks like `oc.getNetworkID()` also does that.

Reply:
oc.getNetworkID() sets it inside `BaseSecondaryNetworkController`
I get it from there and then set it inside `SecondaryLayer3NetworkController`
I know the "oc" doesn't help but :)

Reply:
this was fixed in PR4554